### PR TITLE
Additional information can now be specified for levels.

### PR DESCRIPTION
**Pull request by mrkubax10**

### CODE REVIEW

**Review by slackstone — COMMENTED** [review]
This looks OK to me, seems like a nice addition. I will test the PR on a clean tree build.

File: src/supertux/level_parser.cpp [hunk 1]
```diff
@@ -167,6 +167,7 @@ LevelParser::load(const ReaderDocument& doc)
     level.get("license", m_level.m_license);
     level.get("target-time", m_level.m_target_time);
     level.get("suppress-pause-menu", m_level.m_suppress_pause_menu);
+    level.get("note",m_level.m_note);
```
Comment (tobbi) [inline src/supertux/level_parser.cpp:170]: add a space

File: src/supertux/levelintro.cpp [hunk 2]
```diff
@@ -182,6 +182,11 @@ LevelIntro::draw(Compositor& compositor)
                       Statistics::time_to_string(m_level.m_target_time), targetTimeBeaten);
     }
   }
+  py+=32;
```
Comment (tobbi) [inline src/supertux/levelintro.cpp:185]: spaces around operators and in between arguments

**Review by tobbi — CHANGES_REQUESTED** [review]

File: src/supertux/level_parser.cpp (hunk 1 above)
Comment (mrkubax10) [reply src/supertux/level_parser.cpp]: updated

File: src/supertux/levelintro.cpp (hunk 2 above)
Comment (mrkubax10) [reply src/supertux/levelintro.cpp]: changed

File: data/locale/pl.po [hunk 3]
```diff
@@ -1183,6 +1183,9 @@ msgstr "Wrogowie"
 #: src/supertux/statistics.cpp:271 src/supertux/levelintro.cpp:172
 msgid "Secrets"
 msgstr "Sekrety"
+#: src/supertux/levelintro.cpp:187
+msgid "Level Note"
+msgstr "Informacja o poziome"
```
Comment (Semphriss) [inline data/locale/pl.po:1188]: Please don't add translations manually, use Transifex instead

File: src/audio/sound_manager.cpp [hunk 4]
```diff
@@ -344,11 +344,13 @@ SoundManager::pause_music(float fadetime)
...
     return;
+  
 
   if (fadetime > 0) {
     if (m_music_source
-       && m_music_source->get_fade_state() != StreamSoundSource::FadingPause)
-      m_music_source->set_fading(StreamSoundSource::FadingPause, fadetime);
+       && m_music_source->get_fade_state() != StreamSoundSource::FadingPause){
+        m_music_source->set_fading(StreamSoundSource::FadingPause, fadetime);
+       }
```
Comment (Semphriss) [inline src/audio/sound_manager.cpp:353]: Tab alignment with the closing bracket

File: src/audio/sound_manager.cpp [hunk 5]
```diff
@@ -394,13 +396,15 @@ SoundManager::set_sound_volume(int volume)
...
   if (m_music_source == nullptr)
     return;
 
   if (fadetime > 0) {
     if (m_music_source
-       && m_music_source->get_fade_state() != StreamSoundSource::FadingResume)
-      m_music_source->set_fading(StreamSoundSource::FadingResume, fadetime);
+       && m_music_source->get_fade_state() != StreamSoundSource::FadingResume){
+        m_music_source->set_fading(StreamSoundSource::FadingResume, fadetime);
+       }
```
Comment (Semphriss) [inline src/audio/sound_manager.cpp:407]: Same as above

Why have these lines changes? They don't seem related to the PR...

File: src/supertux/level.cpp [hunk 6]
```diff
@@ -125,6 +126,9 @@ Level::save(Writer& writer)
   writer.write("version", 3);
   writer.write("name", m_name, true);
   writer.write("author", m_author, false);
+  if(!m_note.empty()){
+    writer.write("note",m_note,false);
+  }
```
Comment (Semphriss) [inline src/supertux/level.cpp:131]: Spacing (before open brackets and after commas)

File: src/supertux/levelintro.cpp [hunk 7]
```diff
@@ -182,6 +182,11 @@ LevelIntro::draw(Compositor& compositor)
                       Statistics::time_to_string(m_level.m_target_time), targetTimeBeaten);
     }
   }
+  py += 32;
+  if(!m_level.m_note.empty()){
+    context.color().draw_center_text(Resources::normal_font, _("Level Note")+": "+m_level.m_note, Vector(0,py), LAYER_FOREGROUND1);
+  }
```
Comment (Semphriss) [inline src/supertux/levelintro.cpp:188]: Spacing (same as above, and around operators)

File: src/supertux/menu/editor_level_menu.cpp [hunk 8]
```diff
@@ -34,6 +34,7 @@ EditorLevelMenu::EditorLevelMenu() :
   add_textfield(_("Author"), &(level->m_author));
   add_textfield(_("Contact"), &(level->m_contact));
   add_textfield(_("License"), &(level->m_license));
+  add_textfield(_("Level Note"),(&level->m_note));
```
Comment (Semphriss) [inline src/supertux/menu/editor_level_menu.cpp:37]: Space after comma

Also, ampersand outside parentheses

File: src/supertux/menu/game_menu.cpp [hunk 9]
```diff
@@ -48,7 +48,7 @@ GameMenu::GameMenu() :
   })
 {
   Level& level = GameSession::current()->get_current_level();
-
+
```
Comment (Semphriss) [inline src/supertux/menu/game_menu.cpp:51]: No spaces on empty line

**Review by Semphriss — CHANGES_REQUESTED** [review]
A few more things to change (It seems some unrelated changes slipped in?)

**Review by Semphriss — APPROVED** [review]

File: src/supertux/levelintro.cpp [hunk 10]
```diff
@@ -182,6 +182,11 @@ LevelIntro::draw(Compositor& compositor)
                       Statistics::time_to_string(m_level.m_target_time), targetTimeBeaten);
     }
   }
+  py += 32;
+  if (!m_level.m_note.empty()) {
+    context.color().draw_center_text(Resources::normal_font, _("Level Note") + std::string(": ") + m_level.m_note, Vector(0, py), LAYER_FOREGROUND1);
```
Comment (tobbi) [inline src/supertux/levelintro.cpp:187]: Don't do this. Put *Level Note:  %s* (%s being a placeholder) in one string. Otherwise rtl languages will hate us forever.

I don't know at the top of my head how placeholders work, but you can find other examples in our code.

**Review by tobbi — APPROVED** [review]
Looking good, from what I can see.
